libs/sdk-ui-pivot/src/CorePivotTable.tsx

```diff
@@ -596,347 +413,218 @@ export class CorePivotTablePure extends React.Component<ICorePivotTableProps, IC
 
-        setColumnMaxWidth(columnApi, columnIds, MANUALLY_SIZED_MAX_WIDTH);
-    }
+        if (this.props.execution.fingerprint() === execution.fingerprint()) {
+            this.setState({ error: error.getMessage() });
 
-    private updateAutoResizedColumns = (gridApi: GridApi, columnApi: ColumnApi): void => {
-        this.autoResizedColumns = getAutoResizedColumns(
-            this.tableDescriptor,
-            gridApi,
-            columnApi,
-            this.currentResult,
-            this.containerRef,
-            {
-                measureHeaders: true,
-                padding: 2 * DEFAULT_AUTOSIZE_PADDING + HEADER_CELL_BORDER,
-                separators: this.getSeparators(),
-            },
-        );
-    };
+            onExportReady!(createExportErrorFunction(error));
 
-    private autoresizeAllColumns = async (gridApi: GridApi, columnApi: ColumnApi) => {
-        if (!this.shouldPerformAutoresize() || !this.isColumnAutoresizeEnabled()) {
-            return Promise.resolve();
+            this.props.onError?.(error);
         }
-
-        await sleep(COLUMN_RESIZE_TIMEOUT);
-
-        /*
-         * Ensures correct autoResizeColumns
-         */
-        this.updateAutoResizedColumns(gridApi, columnApi);
-        autoresizeAllColumns(columnApi, this.autoResizedColumns);
     };
 
-    private shouldPerformAutoresize() {
-        if (!this.gridApi) {
-            /*
-             * This is here because of the various timeouts involved in pivot table rendering. Before code
-             * gets here, the table may be in a re-init stage
-             */
-            return false;
-        }
+    //
+    // Table resizing
+    //
 
-        const horizontalPixelRange = this.gridApi.getHorizontalPixelRange();
-        const verticalPixelRange = this.gridApi.getVerticalPixelRange();
+    private growToFit = () => {
+        invariant(this.internal.table);
 
-        return horizontalPixelRange.left === 0 && verticalPixelRange.top === 0;
-    }
+        if (!this.isGrowToFitEnabled()) {
+            return;
+        }
 
-    private isPivotTableReady = (api: GridApi) => {
-        const noRowHeadersOrRows = () => {
-            return Boolean(
-                this.visibleData?.rawData().isEmpty() && this.visibleData.meta().hasNoHeadersInDim(0),
-            );
-        };
-        const dataRendered = () => {
-            return noRowHeadersOrRows() || api.getRenderedNodes().length > 0;
-        };
-        const tablePagesLoaded = () => {
-            const pages = api.getCacheBlockState();
-            return (
-                pages &&
-                Object.keys(pages).every(
-                    (pageId: string) =>
-                        pages[pageId].pageStatus === "loaded" || pages[pageId].pageStatus === "failed",
-                )
-            );
-        };
-        return tablePagesLoaded() && dataRendered();
-    };
+        this.internal.table?.growToFit(this.getResizingConfig());
```

Comment:
Nitpick: unnecessary `?.` we have invariant for `this.internal.table` above.

Reply:
fixed